### ISSUE TITLE
pin-depends get pinned, but not installed

### ISSUE THREAD

**Issue by glennsl** [ocaml/opam#5074]
Given the `opam` file below, running `opam install . --deps-only` will result in `timedesc` getting pinned, but not actually installed. Only after running `opam pin add timedesc https://github.com/daypack-dev/timere.git#8d3f16c` manually will it also get installed.

```
# This file is generated by dune, edit dune-project instead
opam-version: "2.0"
synopsis: "Sensonomic client"
maintainer: ["glennsl"]
authors: ["glennsl"]
depends: [
  "ocaml" {>= "4.13.1"}
  "dune" {>= "2.9" & >= "2.0"}
  "jsoo-react" {dev}
  "js_of_ocaml-ppx" {>= "3.11.0"}
  "brr" {>= "0.0.3"}
  "ppx_jsobject_conv" {>= "0.9.0"}
  "timedesc" {dev}
  "ocamlformat" {= "0.20.1"}
  "merlin" {>= "4.4"}
  "odoc" {with-doc}
]
build: [
  ["dune" "subst"] {dev}
  [
    "dune"
    "build"
    "-p"
    name
    "-j"
    jobs
    "--promote-install-files=false"
    "@install"
    "@runtest" {with-test}
    "@doc" {with-doc}
  ]
  ["dune" "install" "-p" name "--create-install-files" name]
]
pin-depends: [
  [ "timedesc.dev" "git+https://github.com/daypack-dev/timere.git#8d3f16c" ]
  [ "jsoo-react.dev" "git+https://github.com/glennsl/jsoo-react.git#4d3356d" ]
]
```

```
# opam config report
# opam-version         2.1.2 
# self-upgrade         no
# system               arch=x86_64 os=linux os-distribution=garuda os-version=soaring
# solver               builtin-mccs+glpk
# install-criteria     -removed,-count[avoid-version,changed],-count[version-lag,request],-count[version-lag,changed],-count[missing-depexts,changed],-changed
# upgrade-criteria     -removed,-count[avoid-version,changed],-count[version-lag,solution],-count[missing-depexts,changed],-new
# jobs                 23
# repositories         1 (http) (default repo at 1c609c75)
# pinned               2 (git), 1 (rsync)
# current-switch       /home/glennsl/dev/remote/sensonomic/sensonomic/client.ml
# ocaml:native         true
# ocaml:native-tools   true
# ocaml:native-dynlink true
# ocaml:stubsdir       /usr/lib/ocaml/stublibs:/usr/lib/ocaml
# ocaml:preinstalled   true
# ocaml:compiler       system
# ocaml-system:path    /usr/bin
```

**Comment by kit-ty-kate** [comment]
This behaviour still exists in opam master (6ee9209b5b7260f9aca57cc1ff86eeb19828261d).
One thing to note is that this behaviour disappears when the two `{dev}` get removed.

**Comment by AltGr** [comment]
Not obvious, but actually a duplicate of #5177 . Thanks for reporting!
